js/ui/status/automaticUpdates.js

```diff
@@ -151,6 +158,27 @@ var Indicator = new Lang.Class({
         if (!this._client || !this._proxy)
             return;
 
+        if (this._updateTimeoutId > 0) {
+            GLib.source_remove(this._updateTimeoutId);
+            this._updateTimeoutId = 0;
+        }
+
+        // When disconnected, don't timeout before updating
+        if (!this._getActiveConnection()) {
+            this._updateStatus();
+            return;
+        }
+
+        // Use a timeout to avoid instantly throwing the notification at
+        // the users' face, and to avoid a series of unecessary updates
+        // that happen when NetworkManager is still figuring out details.
+        this._updateTimeoutId = GLib.timeout_add_seconds(GLib.PRIORITY_DEFAULT,
+                                                         4,
+                                                         this._updateStatus.bind(this));
```

Comment:
Instead of reusing `_updateStatus` here directly, could you instead use an arrow function that calls `_updateStatus() and then remove the source and set `_updateTimeoutId` to `0`? I don't like adding those extra lines at the end of `_updateStatus` that are only needed because of this use here in `timeout_add_seconds()`.